### PR TITLE
Make reqOption public to allow users of the library to mock it

### PR DESCRIPTION
**Pull request by orishoshan**
Currently, the library can _almost_ be mocked for testing: all of the concrete types used in its interface are data structures, with the exception of reqOption.

### CODE REVIEW

**Comment by orishoshan** [comment]
This PR had previously accidentally included some commits I hadn't meant to put into it that are unrelated to the change in the title - it now includes _only_ the `reqOption` change.

**Comment by alexkappa** [comment]
Hi @orishoshan, thanks for contributing! I understand the use case of mocking out services you include in your own codebase. For what it's worth I made it private on purpose so one would rely on the existing methods for setting options in requests.

I think the changes you propose in this PR are ok, but let me take a better look and get back to you soon.

File: management/management.go [hunk 1]
```diff
@@ -299,20 +299,20 @@ func (m *managementError) Status() int {
 	return m.StatusCode
 }
 
-// reqOption configures a call (typically to retrieve a resource) to Auth0 with
+// ReqOption configures a call (typically to retrieve a resource) to Auth0 with
 // query parameters.
-type reqOption func(url.Values)
+type ReqOption func(url.Values)
```
Comment (alexkappa) [inline management/management.go]: Can we use the name `RequestOption` for this instead? As its becoming part of the public API maybe its okay to be more verbose.

**Comment by alexkappa** [comment]
With v3 this issue is addressed together with support for totals and other improvements.

**Comment by orishoshan** [comment]
Hi @alexkappa,

Sorry for not responding! I forgot I had turned off email notifications for github. The changes made in v3 indeed address this, thank you.

**Pull request closed without merging**